.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,15 +1,24 @@
-#### Because:
-<!--
-If your pull request resolves an open issue, please provide a link to it. For example: Resolves: https://github.com/TheOdinProject/theodinproject/issues/1886
-Otherwise, please briefly explain why these changes were made, what problem does it solve?.
--->
-
-#### This commit
-<!--
-A bullet point list of one or more items outlining what was done in this pull request to solve the problem.
--->
-
-#### Checklist
- - [ ] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/theodinproject/wiki/Contributing-Guide)?
- - [ ] You have verified the tests and linters all pass against your changes?
- - [ ] You have included automated tests for your changes where applicable?
+<!-- Thank you for taking the time to contribute to The Odin Project. In order to get pull requests (PRs) closed in a reasonable amount of time, you must include a baseline of information about the changes you are proposing. Please read this template in its entirety before filling it out to ensure that it is filled out correctly. -->
+
+<!-- Complete the following REQUIRED checkboxes by replacing the whitespace between the square brackets with an 'x', e.g. [x]. -->
+
+-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
+-   [ ] The title of this PR follows the `where the change occurs: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
+
+<!-- Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete these later if they are not currently applicable. -->
+
+-   [ ] I have verified all tests and linters pass against my changes, and I have included automated tests where applicable
+-   [ ] If one exists, I have linked a related open issue to this PR in Step 2 below
+-   [ ] If changes were requested, I have made them and re-requested a review from the maintainer (top of the right sidebar)
+
+**1. Description of the Changes**
```

Comment:
So to give context for this change, it's mainly to have more consistent templates across repos where possible (aside from obvious deviations, like the curriculum repo we don't need users to include automated tests). Basically so users can open a PR without major differences when filling out the template/it's instantly recognizable how it should be filled out.

The reason for using "Description of the Changes" was basically to be a catch-all for both the "why" of the PR and the "how", since it neve really seemed like having those two separated was all that necessary. I can see what you mean by it being *too* broad, though. Beyond that, I don't think I really have any strong opinion either way on format (other than whatever the format,  I think if we can it should be something that can work across a majority of our repos with tweaks mainly to the checkboxes).

What would you think of something along the lines of the following tweaks to the original format:

```
1. Reason for PR
<!-- 
If your PR closes an open issue, link it by including the `Closes` keyword followed by the issue number/issue URL. For example, `Closes #2013`.

Otherwise, please explain the reason for your pull request, e.g. what problem it solves or what benefit it provides.
 -->

2. Outline of Changes
<!-- 
A bullet point list of one or more items outlining what was done in this PR to solve the problem or introduce the feature/enhancement.
 -->
```
Suggesting these tweaks just because if the worry is that users will just mention what's already in the PR, the previous headings on their own didn't really convey what that section should actually include all that well. At least, not without reading the commented out instructions (which the same could be said for the "Description of Changes" instructions below this line).

This would remove the existing "Related issue" step and place it in the first "reason" step, and I think this could be a template that could be used in most of our other repos (which is a big plus in my book).

Reply:
This summarises the approach I use for commit messages: https://thoughtbot.com/blog/5-useful-tips-for-a-better-commit-message

Splitting the why and the how has a bunch of benefits:

* It's easier on the author since the prompts provide clarity with what they need to write which helps keep the message to the point.
* It's easier on the reviewer to get context at a glance as to why the changes in the PR are necessary and how the author resolved the issues. 
* It saves people time in the future. For example, if someone comes across a line of code that they don't understand, or perhaps they just want to know the context as to why it was added in the first place. They can git blame and find out without having to parse the important details from potentially multiple paragraphs of text that open-ended prompts often have.

I've been through a lot of git commit message conventions before landing on this one a few years ago. It's been the best I've come across so far. I'd go so far as to recommend we use it on all repos for the reasons listed above.

I'm not totally against updating the "Because" and "This commit" naming conventions. _But_ looking back at the last few PR's from contributors outside of the maintainer team. It seems like they aren't having issues with filling them in correctly.

Reply:
Definitely understand those benefits. I had been used to the other repos that never were setup with this format, so coming to this format was more jarring for me than vice versa 😆

Like I said, ultimately I don't think I have strong opinions either way re: because/this commit vs reason for/outline of, or any other variation. Ultimately what I'd prefer is a format that, outside of repo-specific checkboxes, is more consistent and easily recognizable regardless of the repo for any experience level of user (obviously we can't capture *every* user, though).

If you believe the "because/this commit" format coudl hit that nail,, I'd be down for implementing this sort of format across other repos as well. Otherwise, if you feel this format is more beneficial to this repo only, I'd also be down to just having this main repo use one PR template and the other repos use another (because this repo is much more different than the other ones in terms of content).

Reply:
Any repos with a codebase can definitely benefit from it like this one and the exercise repos. I think the curriculum can as well. Most of the PR's we get are tweaks to the existing content so having structured explanations for changes made could be immensely helpful.

I've been using this format on the curriculum repo 😆
https://github.com/TheOdinProject/curriculum/pull/23212
https://github.com/TheOdinProject/curriculum/pull/23219

Reply:
So the last thing I think to clarify (don't quote me on this later 😆 ), is the issue/PR title format. This PR uses a "location of change: description" format, whereas I've noticed you use more of a "keyword: description" format.

The reason I had opted for the "location" of the change is to let contributors know immediatley where an issue/PR pertains to, which along with the description should help avoid duplicates being made. So using your two examples above with this format, it would have ended up being "Foundations: Archive old Frontend lessons" and "Intermediate HTML section: Add Form Basics lessons".

It may not need to be an either/or situation, as maybe both the keyword and location could work together (e.g. `Feature - Intermediate HTML: Add Form Basics lesson`), but let me know what you think.

Reply:
I _really_ like the `location of change: description` format for the curriculum, I think that will be a great system. It makes a lot of sense and I can definitely see its utility in being able to spot duplicated issues.

But I don't think it would work as well when changes will be mostly code related as they will be in this and the exercise repos. Just because the location of change would mostly pertain to files and there may be many files changed from different parts of the app in any given PR.

I'd prefer we use scrum story types in this repo please:
* Feature - adds new or amends user-facing behaviour
* Chore - changes that have no user-facing value, refactors, dependency bumps etc.
* Fix - Bug fixes.

Reply:
I definitely wasn't gonna follow up with clarification on what you view a "chore" as, as I wasn't sure if the chore label description is all that accurate 😆 

Sounds good, that difference definitely makes sense. I'll update things and re-request your review. Thanks for all the input!

Reply:
Sounds good, thanks for putting this all together @thatblindgeye 💪